provider/os/os_discover.go

```diff
@@ -128,16 +128,16 @@ func newClient(args map[string]string, l *log.Logger) (*gophercloud.ServiceClien
 	}
 	projectID := argsOrEnv(args, "project_id", "OS_PROJECT_ID")
 	insecure := argsOrEnv(args, "insecure", "OS_INSECURE")
+	domain_id := argsOrEnv(args, "domain_id", "OS_DOMAIN_ID")
+	domain_name := argsOrEnv(args, "domain_name", "OS_DOMAIN_NAME")
 
 	if url == "" {
 		return nil, fmt.Errorf("discover-os: Auth url must be provided")
 	}
 
 	ao := gophercloud.AuthOptions{
-		// "domain_id": OS_DOMAIN_ID
```

Comment:
Interestingly, so previously we just...didn't set this and it still worked? Do you mind explaining this at all? The change seems ok to me but given we have no openstack infrastructure we can't run full end-to-end tests against it.

Reply:
Got it, yeah, seems safe to me.